flytekit/models/common.py

```diff
@@ -40,6 +44,29 @@ def from_flyte_idl(cls, idl_object):
         pass
 
 
+def _repr_idl_yaml_like(idl, indent=0) -> str:
```

Comment:
Why not make this the default repr

Reply:
I wanted to make it the default, but I saw the current repr strips away the newlines:

https://github.com/flyteorg/flytekit/blob/e9f349910e660f25f5e33e360ea16813099b27d6/flytekit/models/common.py#L63

Was there a reason for making it all one line?

Reply:
i don't think so, not a good one anyways.  feel free to overwrite.  the reason likely was logging platforms (cloudwatch logs, splunk, grafana, etc) don't render logs with new lines well.  but i don't know how much this could matter for python.  that's more of a backend issue.